core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -76,7 +76,7 @@
 
     @Override
     public <ContainerType> ContainerType collectInto(Class<ContainerType> containerType) {
-        return collect(getCollectorFactoryRegistry().createCollectorFor(containerType));
+        return collect(getCollectorFactoryRegistry().createCollectorFor(getHandle().getTypeResolver().resolve(containerType)));
```

Comment:
I'm starting to think this `collectInto(Class)` method isn't buying us much. You can already use `collect(toSet())` and, because of the generics signature of `Collector.toSet()` is set up, we're guarantee to get back a `Set<ResultType>`.